src/util/psbt/map/input.rs

```diff
@@ -55,6 +55,15 @@ pub struct Input {
     /// The finalized, fully-constructed scriptWitness with signatures and any
     /// other scripts necessary for this input to pass validation.
     pub final_script_witness: Option<Vec<Vec<u8>>>,
+    /// TODO: Proof of reserves commitment
+    /// RIPEMD hash to preimage map
+    pub ripemd_preimages: BTreeMap<ripemd160::Hash, Vec<u8>>,
```

Comment:
If new hash types get added there needs to be a new `BTreeMap` here. Would there be a downside to having one `BTreeMap<PsbtHash, Vec<u8>>` instead?

Reply:
I considered this previously and it is a design decision. We can do that the way you suggested but I initially chose this way because it made sense to follow the fields exactly as written in the BIP document. 

I don't know if there would implementation complexity elsewhere because of using one map. But considering we won't likely be having more hash types be added here. I think keeping it as it is makes things easier.